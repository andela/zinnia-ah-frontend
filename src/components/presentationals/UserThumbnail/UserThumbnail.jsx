import React from 'react';
import PropTypes from 'prop-types';

//style
import 'semantic-ui-css/semantic.min.css';
import './UserThumbnail.scss';

import Avatar from '../Avatar/Avatar';

const UserThumbnail = ({ url, image, name, info }) => (
  <a className="user-thumbnail" href={`@${url}`}>
    <Avatar url={image} className="avatar-m" />
    <div>
      <span className="name"> {name} </span>
      <span className="info"> {info} </span>
    </div>
  </a>
);

UserThumbnail.propTypes = {
  url: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,
};

export default UserThumbnail;
