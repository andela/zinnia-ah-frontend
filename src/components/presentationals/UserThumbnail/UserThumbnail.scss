.user-thumbnail {
  align-items: center;
  color: #000;
  display: flex;
  flex-basis: 70%;
  flex-direction: row;
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  justify-content: left;
  .avatar {
    margin-right: 10px;
  }

  span {
    display: block;
    .name {
      font-size: 16px;
      line-height: 17px;
    }
  }

  .info {
    opacity: 0.6;
  }
}
