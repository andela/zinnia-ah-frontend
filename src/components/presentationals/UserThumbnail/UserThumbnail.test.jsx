import React from 'react';
import { mount } from 'enzyme';
import checkPropTypes from 'check-prop-types';

import UserThumbnail from './UserThumbnail';

const imageSample = {
  url: 'http://example.com/randomuser',
  name: 'sample name',
  image: 'image_url.png',
  info: '@sample_name',
};

describe('UserThumbnail', () => {
  it('should render UserThumbnail correctly', () => {
    const component = mount(<UserThumbnail {...imageSample} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should receive props', () => {
    const propsError = checkPropTypes(
      UserThumbnail.propTypes,
      imageSample,
      'props',
      UserThumbnail.imageSample,
    );
    expect(propsError).toBeUndefined();
  });
});
