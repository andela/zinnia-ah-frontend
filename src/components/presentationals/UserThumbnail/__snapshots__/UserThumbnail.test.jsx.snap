// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserThumbnail should render UserThumbnail correctly 1`] = `
<UserThumbnail
  image="image_url.png"
  info="@sample_name"
  name="sample name"
  url="http://example.com/randomuser"
>
  <a
    className="user-thumbnail"
    href="@http://example.com/randomuser"
  >
    <Avatar
      className="avatar-m"
      url="image_url.png"
    >
      <img
        alt="avatar"
        className="avatar avatar-m"
        src="image_url.png"
      />
    </Avatar>
    <div>
      <span
        className="name"
      >
         
        sample name
         
      </span>
      <span
        className="info"
      >
         
        @sample_name
         
      </span>
    </div>
  </a>
</UserThumbnail>
`;
